File: queue/Q933.java
import java.util.Deque;
import java.util.LinkedList;

public class Q933 {
    /**
     * Your RecentCounter object will be instantiated and called as such:
     * RecentCounter obj = new RecentCounter();
     * int param_1 = obj.ping(t);
     */
    private static class RecentCounter {
        private final Deque<Integer> deque = new LinkedList<>();

        public RecentCounter() {

        }

        public int ping(int t) {
            deque.addLast(t);
            while (!deque.isEmpty() && deque.getFirst() < t - 3000) {
                deque.removeFirst();
            }
            return deque.size();
        }
    }
}

File: LinkedList-Stack-Queue/Q933.py
# https://leetcode.com/problems/number-of-recent-calls/

from collections import deque


class RecentCounter:
    def __init__(self):
        self.times = deque()
        self.interval = 3000  # milliseconds

    def ping(self, time: int) -> int:
        self.times.append(time)
        while (self.times[-1] - self.times[0]) > self.interval:
            self.times.popleft()
        return len(self.times)


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)
